PostHog/Replay/UIView+Util.swift

```diff
@@ -34,17 +34,25 @@
         }
 
         func toImage() -> UIImage? {
-            let size = bounds.size
+            // Avoid Rendering Offscreen Views
+            let bounds = superview?.bounds ?? bounds
+            let size = bounds.intersection(bounds).size
 
             if !size.hasSize() {
                 return nil
             }
 
-            let renderer = UIGraphicsImageRenderer(size: size)
+            let rendererFormat = UIGraphicsImageRendererFormat.default()
 
-            let image = renderer.image { context in
-                // Render the view's layer into the current context
-                layer.render(in: context.cgContext)
+            // This can significantly improve rendering performance because the renderer won't need to
+            // process transparency.
+            rendererFormat.opaque = isOpaque
+            // Another way to improve rendering performance is to scale the renderer's content.
+            // rendererFormat.scale = 0.5
+            let renderer = UIGraphicsImageRenderer(size: size, format: rendererFormat)
+
+            let image = renderer.image { _ in
+                drawHierarchy(in: bounds, afterScreenUpdates: false)
```

Comment:
This feels like a big improvement compared to `layer.render(in: context.cgContext)`